odoo/addons/base/models/res_currency.py

```diff
@@ -255,12 +256,14 @@ def is_zero(self, amount):
         return tools.float_is_zero(amount, precision_rounding=self.rounding)
 
     @api.model
-    def _get_conversion_rate(self, from_currency, to_currency, company, date):
-        currency_rates = (from_currency + to_currency)._get_rates(company, date)
-        res = currency_rates.get(to_currency.id) / currency_rates.get(from_currency.id)
-        return res
-
-    def _convert(self, from_amount, to_currency, company, date, round=True):
+    def _get_conversion_rate(self, from_currency, to_currency, company=None, date=None):
+        if from_currency == to_currency:
+            return 1
+        company = company or self.env.company
+        date = date or fields.Date.context_today(self)
+        return from_currency.with_company(company).with_context(to_currency=to_currency.id, date=str(date)).rate
```

Comment:
(minor) I don't think that stringifying `date` is necessary, unless that context is aimed at being sent/returned by RPC.

Reply:
I don't want to compute the value for both `'2010-10-10'` and `date(2010, 10, 10)`, I think it is just safer generally :shrug: